### PR TITLE
feat(ui): Add Template/Cron workflow filter to workflow page. Closes #4532

### LINKED ISSUE #4532: Add filter all cronWorkflow/templateWorkflow to UI Workflow labels filter
# Summary

What change needs making?

Add `workflows.argoproj.io/cron-workflow`, `workflows.argoproj.io/workflow-template` to the UI Labels autocomplete

For example,
tempsnip

# Use Cases

When would you use this?

This will enable us to quickly filter all cronWorkflows or templateWorkflows.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by tczhao**
Signed-off-by: Tianchu Zhao evantczhao@gmail.com

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

Closes #4532

image

### CODE REVIEW

File: ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx [hunk 1]
```diff
@@ -33,13 +63,25 @@ export class WorkflowFilters extends React.Component<WorkflowFilterProps, {}> {
...
-                            autocomplete={this.getLabelSuggestions(this.props.workflows)}
+                            autocomplete={this.labelSuggestion}
                             tags={this.props.selectedLabels}
                             onChange={tags => {
                                 this.props.onChange(this.props.namespace, this.props.selectedPhases, tags);
                             }}
                         />
                     </div>
+                    <div className='columns small-3 xlarge-12'>
+                        <p className='wf-filters-container__title'>Workflow Template</p>
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx:74]: I think we should extract these two dropdowns into a shared component.

**Review by alexec — CHANGES_REQUESTED** [review]
Looks good.

* Do you think you could extract the drop-downs into shared code so we can reuse them?
* Can I ask for screenshots to be attached?

File: ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx (hunk 1 above)
Comment (tczhao) [reply ui/src/app/workflows/components/workflow-filters/workflow-filters.tsx:62]: - added screenshot
- added dropdown share component

**Comment by tczhao** [comment]
Hi, I've added the organization.
I'm not sure if the LinkedIn link from my GitHub profile is enough to verify that I work for the org.
Otherwise, I can send you a message using org email from slack to prove it.

File: ui/src/app/shared/components/dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { DataLoader, Select, SelectOption } from 'argo-ui';
+
+interface Props {
+    load: Promise<(string | SelectOption)[]>;
+    onChange: (value: string) => void;
+}
+
+export class Dropdown extends React.Component<Props, { selected: string }> {
```
Comment (alexec) [inline ui/src/app/shared/components/dropdown.tsx:9]: Can I get you to change this into a stateless component? They are much more robust and I think we want to move away from class components.

https://reactjs.org/docs/hooks-intro.html

Comment (alexec) [reply ui/src/app/shared/components/dropdown.tsx:9]: >  functional component instead

Comment (tczhao) [reply ui/src/app/shared/components/dropdown.tsx]: updated to functional component

File: ui/src/app/shared/components/dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { DataLoader, Select, SelectOption } from 'argo-ui';
+
+export const Dropdown = (props: {load: Promise<(string | SelectOption)[]>; onChange: (value: string) => void}) => {
```
Comment (alexec) [inline ui/src/app/shared/components/dropdown.tsx:4]: Have about calling this `DataLoaderDropdown`?

**Review by alexec — CHANGES_REQUESTED** [review]
One last change.

File: ui/src/app/shared/components/dropdown.tsx (hunk 3 above)
Comment (tczhao) [reply ui/src/app/shared/components/dropdown.tsx:4]: How about the filename?

Comment (alexec) [reply ui/src/app/shared/components/dropdown.tsx:4]: `data-loader-dropdown.tsx`

Comment (tczhao) [reply ui/src/app/shared/components/dropdown.tsx]: updated to `DataLoaderDropdown`

**Comment by alexec** [comment]
Please run `yarn lint` to fix lint errors.

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
THank you!

**Pull request merged**